wire-ios-request-strategy/Sources/APIs/MessageAPI.swift

```diff
@@ -265,6 +271,87 @@ class MessageAPIV4: MessageAPIV3 {
     override var apiVersion: APIVersion {
         .v4
     }
+
+    private let protobufContentType = "application/x-protobuf"
+
+    override func broadcastProteusMessage(message: any ProteusMessage) async throws -> (Payload.MessageSendingStatus, ZMTransportResponse) {
+        let path = "/broadcast/proteus/messages"
+
+        guard let encryptedPayload = await message.encryptForTransportQualified() else {
+            WireLogger.messaging.error("failed to encrypt message for transport")
+            throw NetworkError.errorEncodingRequest
+        }
+
+        let request = ZMTransportRequest(
+            path: path,
+            method: .post,
+            binaryData: encryptedPayload.data,
+            type: protobufContentType,
+            contentDisposition: nil,
+            apiVersion: apiVersion.rawValue
+        )
+
+        let response = await httpClient.send(request)
+
+        if response.httpStatus == 412 {
+            // New V4 payload
+            guard let messageSendingStatus = Payload.MessageSendingStatusV4(
+                response,
+                decoder: .defaultDecoder
+            ) else {
+                throw NetworkError.errorDecodingResponse(response)
+            }
+
+            throw NetworkError.missingClients(messageSendingStatus.toAPIModel(), response)
+
+        } else {
+            let payload: Payload.MessageSendingStatusV4 = try mapResponse(response)
+            return (payload.toAPIModel(), response)
+        }
```

Comment:
optional: Personally I would extract this duplicate code from these two methods to a local private method although I'm sure there are arguments against making this too dry.

Reply:
Yes I would too, but this `MessageAPI` will be replaced with something in `WireAPI` so I don't think it's worth doing just for the short term.

Reply:
Cool. Makes sense.